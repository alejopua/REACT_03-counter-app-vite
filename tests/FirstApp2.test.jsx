const { render, screen } = require("@testing-library/react");
const { FirstApp } = require("../src/FirstApp");

describe('Testing in <FirstApp />', () => {
  const principalTitle = 'Hi, I am Alejandro';
  const title = 'I am a Software Developer';
  
  test('should match with snapshot', () => {
    const { container } = render( <FirstApp principalTitle={ principalTitle } /> )
    expect( container ).toMatchSnapshot();
  });

  test('must should message "Hi, I am Alejandro"', () => {
    render( <FirstApp principalTitle={ principalTitle } /> )
    expect( screen.getByText(principalTitle) ).toBeTruthy();
    // screen.debug();
  });

  test('must should the principalTitle in a h1', () => {
    render( <FirstApp principalTitle={ principalTitle } /> )
    expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain(principalTitle);
    // screen.debug();
  });

  test('must show the title in component', () => {
    render(
      <FirstApp 
        principalTitle={principalTitle} 
        title={title}
      />
    )
  
    expect( screen.getAllByText(title).length).toBe(2);
  });

});