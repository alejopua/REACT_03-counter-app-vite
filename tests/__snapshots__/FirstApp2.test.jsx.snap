// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <FirstApp /> should match with snapshot 1`] = `
<div>
  <h1
    class="text-6xl font-bold text-teal-50"
    data-testid="test-principalTitle"
  >
    Hi, I am Alejandro
  </h1>
  <h2
    class="text-6xl font-bold text-teal-50"
  >
    No title
  </h2>
  <h3
    class="text-6xl font-bold text-teal-50"
  >
    No title
  </h3>
  <h4
    class="text-3xl text-teal-50"
  >
    No sub-title
  </h4>
</div>
`;
